qemu/tests/uefishell.py

```diff
@@ -121,21 +123,28 @@ def send_command(self, command, check_result=False, interval=0.5):
 
         :param command: the command string being executed
         :param check_result: the pattern to validate output
+        :param interval: time interval between commands
+        :return if check_result is True, return matched string
         """
         logging.info("Send uefishell command: %s" % command)
         output = self.session.cmd_output_safe(command)
         time.sleep(interval)
-        #Judge if cmd is run successfully via environment variable 'lasterror'
+        # Judge if cmd is run successfully via environment variable 'lasterror'
         last_error = self.params["last_error"]
         env_var = self.session.cmd_output_safe("set")
-        if not re.search(last_error, env_var):
+        if not re.search(last_error, env_var, re.S):
             self.test.fail("Following errors appear %s when running command %s"
                            % (output, command))
         if check_result:
+            value = ""
             for result in check_result.split(", "):
-                if not re.search(result, output):
+                if not re.findall(result, output, re.S):
                     self.test.fail("The command result is: %s, which does not"
-                                   " match the expectation: %s" % (output, result))
+                                   " match the expectation: %s"
+                                   % (output, result))
+                else:
+                    value += re.findall(result, output, re.S)[0]
```

Comment:
check_result can contain several keywords/patterns to be checked, value += xxx just simply join the matched together, so is it better to return a list?

Reply:
agree with you. updated it. thanks